File: resolve_conflicts.py
import re
import os

files_to_fix = [
    r'src\declarations.d.ts',
    r'src\screens\loginscreen.tsx',
    r'package-lock.json'
]

base_dir = r'd:\react-native cli\userPng'

def resolve_file(filepath):
    full_path = os.path.join(base_dir, filepath)
    if not os.path.exists(full_path):
        print(f"File not found: {full_path}")
        return

    with open(full_path, 'r', encoding='utf-8') as f:
        content = f.read()

    # Regex to find merge conflict blocks
    # Captures:
    # 1. Content between HEAD and ======= (Current Change)
    # 2. Content between ======= and >>>>>>> (Incoming Change)
    # We want to keep group 2.
    
    # Pattern explanation:
    # <<<<<<< HEAD\n   or <<<<<<< [^\n]+\n
    # (.*?)            Content A (lazy)
    # =======\n
    # (.*?)            Content B (lazy)
    # >>>>>>> [^\n]+\n
    
    pattern = re.compile(r'<<<<<<< [^\n]+\n(.*?)=======\n(.*?)>>>>>>> [^\n]+\n', re.DOTALL)
    
    # Function to replace with incoming content
    def replace_with_incoming(match):
        return match.group(2)

    new_content, count = pattern.subn(replace_with_incoming, content)
    
    if count > 0:
        print(f"Resolved {count} conflicts in {filepath}")
        with open(full_path, 'w', encoding='utf-8') as f:
            f.write(new_content)
    else:
        print(f"No conflicts found in {filepath}")

for file in files_to_fix:
    resolve_file(file)

File: resolve_conflicts.js
const fs = require('fs');
const path = require('path');

const files = [
    'src/declarations.d.ts',
    'src/screens/loginscreen.tsx',
    'package-lock.json'
];

const baseDir = '.';

files.forEach(file => {
    const fullPath = path.join(baseDir, file);
    if (fs.existsSync(fullPath)) {
        const content = fs.readFileSync(fullPath, 'utf8');
        // Regex: <<<<<<< .*\n([\s\S]*?)=======\n([\s\S]*?)>>>>>>> .*\n
        // Replace with group 2

        // We match explicitly the conflict markers
        const pattern = /<<<<<<< [^\n]+\r?\n([\s\S]*?)=======\r?\n([\s\S]*?)>>>>>>> [^\n]+(\r?\n)?/g;

        let count = 0;
        const newContent = content.replace(pattern, (match, p1, p2) => {
            count++;
            return p2;
        });

        if (count > 0) {
            fs.writeFileSync(fullPath, newContent, 'utf8');
            console.log(`Resolved ${count} conflicts in ${file}`);
        } else {
            console.log(`No conflicts found in ${file}`);
        }
    } else {
        console.log(`File not found: ${file}`);
    }
});
